src/data.c

```diff
@@ -380,7 +382,7 @@ void fill_truth_detection(const char *path, int num_boxes, float *truth, int cla
             ++sub;
             continue;
         }
-        if ((w < lowest_w || h < lowest_h)) {
+        if ((w < lowest_w || h < lowest_h) || (area < min_area)) {
```

Comment:
lowest_w and lowest_h are both calculated relative to the network size.

Interested in why you've used a constant (w.r.t. network size) float, rather than defining it as the (integer) area in input pixels.

Reply:
`area` is a threshold on the annotation area which is expressed in "yolo" units. Hence, this is independent from the network resolution and it is related to the image size

In my dataset, I mainly use 1920x1080px images, and a `min_area=0.0001` will ignore all the annotations smaller than 14x14px. These are arguably useful for learning, but I can reintroduce them by changing `min_area` without touching the annotations.

Reply:
A bit more detail on my thinking..

There may be two reasons for ignoring these small annotations:

a) Because of human error annotating the data.
b) Because we assume there is not enough information to accurately detect an object with a bounding area of `<min_area`

if (a) is true, then more data should surely fix this (law of large numbers etc). If we really want to remove these then this a trivial quick data-cleanup operation on our training images prior to learning.

if (b) is the reason, then shouldn't this filter be applied **after data augmentation**, based on the amount of data available during training - rather than the amount of data in the input image?

Reply:
I agree that is trivial to fix case **a)** through some data cleaning, but I found that different architectures (_v2_, _v3_, _tiny v3_, etc...) at different resolutions may require a slightly different `min_area` threshold. I suspect this may indicate that the amount of information in a single annotation may be or not relevant depending on the net we want to train. Hence, I'd be worried of discarding a-priori some annotations only because a certain net couldn't learn anything from it.

This, in turn, links to **b)**.

>  shouldn't this filter be applied after data augmentation, based on the amount of data available during training - rather than the amount of data in the input image?

Probably this would be viable in case of fixed augmentation, i.e., the images are scaled all the times in the same manner. This is not darkness case as I'm showing in the first comment to this PR.

I hope this make sense :)